### ISSUE TITLE
Performance issues in /data_provider (by P3)

### ISSUE THREAD

**Issue by DLPerf** [MaybeShewill-CV/attentive-gan-derainnet#82]
Hello! I've found a performance issue in /data_provider/data_feed_pipline.py: `batch()` should be called before `map()`, which could make your program more efficient. Here is [the tensorflow document](https://tensorflow.google.cn/guide/data_performance?hl=zh_cn#vectorized_mapping) to support it.

Detailed description is listed below:

- `dataset.batch(batch_size, drop_remainder=True)`[(here)](https://github.com/MaybeShewill-CV/attentive-gan-derainnet/blob/4ec79993cf3d757741c9c88c1036d87bd46e982d/data_provider/data_feed_pipline.py#L290) should be called before `dataset.map(map_func=tf_io_pipline_tools.decode,num_parallel_calls=CFG.TRAIN.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/attentive-gan-derainnet/blob/4ec79993cf3d757741c9c88c1036d87bd46e982d/data_provider/data_feed_pipline.py#L270), `dataset.map(map_func=tf_io_pipline_tools.augment_for_train,num_parallel_calls=CFG.TRAIN.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/attentive-gan-derainnet/blob/4ec79993cf3d757741c9c88c1036d87bd46e982d/data_provider/data_feed_pipline.py#L273), `dataset.map(map_func=tf_io_pipline_tools.augment_for_test,num_parallel_calls=CFG.TRAIN.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/attentive-gan-derainnet/blob/4ec79993cf3d757741c9c88c1036d87bd46e982d/data_provider/data_feed_pipline.py#L276) and `dataset.map(map_func=tf_io_pipline_tools.normalize,num_parallel_calls=CFG.TRAIN.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/attentive-gan-derainnet/blob/4ec79993cf3d757741c9c88c1036d87bd46e982d/data_provider/data_feed_pipline.py#L278).

Besides, you need to check the function called in `map()`(e.g., `tf_io_pipline_tools.normalize` called in `dataset.map(map_func=tf_io_pipline_tools.normalize,num_parallel_calls=CFG.TRAIN.CPU_MULTI_PROCESS_NUMS)`) whether to be affected or not to make the changed code work properly. For example, if `tf_io_pipline_tools.normalize` needs data with shape (x, y, z) as its input before fix, it would require data with shape (batch_size, x, y, z).

Looking forward to your reply. Btw, I am very glad to create a PR to fix it if you are too busy.
